x-pack/plugins/cases/public/components/actions/assignees/edit_assignees_flyout.tsx

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo, useState } from 'react';
+import { css } from '@emotion/react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  euiFullHeight,
+  EuiLoadingSpinner,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+
+import type { UserProfileWithAvatar } from '@kbn/user-profile-components';
+import { useBulkGetUserProfiles } from '../../../containers/user_profiles/use_bulk_get_user_profiles';
+import type { Case } from '../../../../common';
+import { EditAssigneesSelectable } from './edit_assignees_selectable';
+import * as i18n from './translations';
+import type { ItemsSelectionState } from '../types';
+
+interface Props {
+  selectedCases: Case[];
+  onClose: () => void;
+  onSaveAssignees: (args: ItemsSelectionState) => void;
+}
+
+const fullHeight = css`
+  ${euiFullHeight()}
+
+  .euiFlyoutBody__overflowContent {
+    ${euiFullHeight()}
+  }
+`;
+
+const getUnknownUsers = (
+  assignees: Set<string>,
+  userProfiles?: Map<string, UserProfileWithAvatar>
+) => {
+  const unknownUsers: string[] = [];
+
+  if (!userProfiles) {
+    return unknownUsers;
+  }
+
+  for (const assignee of assignees) {
+    if (!userProfiles.has(assignee)) {
+      unknownUsers.push(assignee);
+    }
+  }
+
+  return unknownUsers;
+};
+
+const EditAssigneesFlyoutComponent: React.FC<Props> = ({
+  selectedCases,
+  onClose,
+  onSaveAssignees,
+}) => {
+  const assignees = useMemo(
+    () => new Set(selectedCases.map((theCase) => theCase.assignees.map(({ uid }) => uid)).flat()),
+    [selectedCases]
+  );
+
+  const { data: userProfiles, isFetching: isLoadingUserProfiles } = useBulkGetUserProfiles({
```

Comment:
nit: Does this and line 79 need to be here? It feels like logic that belongs inside `EditAssigneesSelectable`.

We don't use `userProfiles`, or `unknownUsers` for anything here.

```
{isLoadingUserProfiles ? (
  <EuiLoadingSpinner />
) : (
```

Could also be inside `EditAssigneesSelectable`.

Reply:
I followed the pattern I did with tags. I think the reason I did it is to be able to use other user profiles and not only the ones that come from the bulk get. You are right, though, it makes sense to be inside the selectable. I will do it.